nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/flow/StandardStatelessDataflowFactory.java

```diff
@@ -167,7 +163,9 @@ private synchronized PropertyEncryptor getEncryptor() {
                         return created;
                     }
 
-                    created = getPropertyEncryptor(engineConfiguration.getSensitivePropsKey());
+                    created = new PropertyEncryptorBuilder(engineConfiguration.getSensitivePropsKey())
```

Comment:
What do you think about making the algorithm provided by the stateless properties instead of hard coding it here?  I suppose this could always be added later if we wanted to be more consistent with NiFi, and perhaps the use case is already so small for sensitive props key that hard-coding is acceptable here.  However, I wanted to pose the question here.

Reply:
That's a good question.

I raised a similar question on the initial implementation PR, and at the time, the goal was to keep Stateless as simple as possible. Given that the previous default value in `nifi.properties` remained unchanged for many releases, making this configurable does not seem to provide a great deal of value.

In the future, it may also make sense to deprecate and remove the ability to configure this in `nifi.properties`, and instead prefer a sensible default value. This could be addressed in a separate issue if there are other reasons to make it configurable.